config/initializers/warden.rb

```diff
@@ -75,4 +75,26 @@ def call(env)
   end
 end
 
+Warden::Proxy.class_eval do
+  ENV_SESSION_OPTIONS = 'rack.session.options'.freeze
+  def set_user(user, opts = {})
+    scope = (opts[:scope] ||= @config.default_scope)
+
+    # Get the default options from the master configuration for the given scope
+    opts = (@config[:scope_defaults][scope] || {}).merge(opts)
+    opts[:event] ||= :set_user
+    @users[scope] = user
+
+    if opts[:store] != false && opts[:event] != :fetch
+      options = env[ENV_SESSION_OPTIONS]
+      session_serializer.store(user, scope)
+    end
```

Comment:
For our future reference, `set_user` is copy/pasta from https://github.com/hassox/warden/blob/cb09b959bcac82a087ff0f2938273e6924d845d8/lib/warden/proxy.rb#L172-L176 except:

```diff
   if opts[:store] != false && opts[:event] != :fetch
     options = env[ENV_SESSION_OPTIONS]
-    options[:renew] = true if options
     session_serializer.store(user, scope)
   end
```

I know we're considering a switch to OmniAuth (#157), but I wonder if we should submit a patch to Warden along the lines of:

```diff

```diff
   if opts[:store] != false && opts[:event] != :fetch
     options = env[ENV_SESSION_OPTIONS]
-    options[:renew] = true if options
+    options[:renew] = true if options && !opts[:renew]
     session_serializer.store(user, scope)
   end
```